ow_lander/scripts/all_action_trajectories.py

```diff
@@ -0,0 +1,853 @@
+    
+    plan_c = change_joint_value(move_arm, cs, start_state, constants.J_HAND_YAW,  math.pi/2.2)
+    
+    circ_traj = cascade_plans (circ_traj, plan_c)
+    
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, circ_traj, move_arm, moveit_fk)
+    ##if not parallel:
+    ## Once aligned to trench goal, place hand above trench middle point
+    z_start = ground_position + constants.R_PARALLEL_FALSE_A #- depth
+    
+    plan_d = go_to_Z_coordinate(move_arm, cs, end_pose, x_start, y_start, z_start)
+    circ_traj = cascade_plans (circ_traj, plan_d)
+    
+    ## Rotate hand perpendicular to arm direction
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, circ_traj, move_arm, moveit_fk)
+    plan_e = change_joint_value(move_arm, cs, start_state, constants.J_HAND_YAW, -0.29*math.pi)
+    circ_traj = cascade_plans (circ_traj, plan_e)
+    
+  else:
+      
+    plan_a= move_to_pre_trench_configuration(
+      move_arm, robot, x_start, y_start) 
+    if len(plan_a.joint_trajectory.points) == 0:  # If no plan found, abort
+      return False
+    # Rotate hand so scoop is in middle point
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, plan_a, move_arm, moveit_fk)
+    plan_b = change_joint_value(move_arm, cs, start_state, constants.J_HAND_YAW, 0.0)
+    circ_traj = cascade_plans (plan_a, plan_b)
+    
+    # Rotate scoop
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, circ_traj, move_arm, moveit_fk)
+    plan_c = change_joint_value(move_arm, cs, start_state, constants.J_SCOOP_YAW, math.pi/2)
+    circ_traj = cascade_plans (circ_traj, plan_c)
+    
+    # Rotate dist so scoop is back
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, circ_traj, move_arm, moveit_fk)
+    plan_d = change_joint_value(move_arm, cs, start_state, constants.J_DIST_PITCH, -19.0/54.0*math.pi)
+    circ_traj = cascade_plans (circ_traj, plan_d)
+    
+    # Once aligned to trench goal, place hand above trench middle point
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, circ_traj, move_arm, moveit_fk)
+    z_start = ground_position + constants.R_PARALLEL_FALSE_A - depth
+    plan_e = go_to_XYZ_coordinate(move_arm, cs, end_pose, x_start, y_start, z_start)
+    circ_traj = cascade_plans (circ_traj, plan_e)
+    
+    # Rotate dist to dig
+    cs, start_state, end_pose = calculate_joint_state_end_pose_from_plan_arm (robot, circ_traj, move_arm, moveit_fk)
+    dist_now = start_state[3]
+    plan_f = change_joint_value(move_arm, cs, start_state, constants.J_DIST_PITCH, dist_now + 2*math.pi/3)
+    circ_traj = cascade_plans (circ_traj, plan_f)
+
+  return circ_traj
+
+
+dig_linear_traj = RobotTrajectory()
+
+
+def calculate_joint_state_end_pose_from_plan_arm (robot, plan, move_arm, moveit_fk):
```

Comment:
Every line that calls this function is quite long.  Suggest shortening the name, e.g. `joint_state_end_pose` or even `end_pose` if the rest is inferable by context.

Reply:
the long name specifies what the function is doing....